frontend/internal-packages/agent/src/db-agent/tools/schemaDesignTool.ts

```diff
@@ -30,23 +106,80 @@ export const schemaDesignTool = tool(
       return `Input validation failed: ${errorDetails}. Please check your operations format and ensure all required fields are provided correctly.`
     }
 
-    const result = await repositories.schema.createVersion({
+    // Get current schema to validate DDL before creating version
+    const schemaResult = await repositories.schema.getSchema(designSessionId)
+    if (schemaResult.isErr()) {
+      // LangGraph tool nodes require throwing errors to trigger retry mechanism
+      // eslint-disable-next-line no-throw-error/no-throw-error
+      throw new Error(
+        'Could not retrieve current schema for DDL validation. Please check the schema ID and try again.',
+      )
+    }
+
+    // Apply operations to current schema to get the updated schema
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const currentSchema = structuredClone(schemaResult.value.schema) as Schema
+    applySchemaOperations(currentSchema, parsed.output.operations)
+
+    // Validate DDL by generating and executing it
+    const { ddlStatements, results } = await validateAndExecuteDDL(
+      currentSchema,
+      designSessionId,
+    )
```

Comment:
Perhaps the base DDL with initial_schema_snapshot should be run first, and the DDL should be run here only for the operation. This will be fixed in the next PR.